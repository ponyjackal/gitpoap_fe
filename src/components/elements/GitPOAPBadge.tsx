import React from 'react';
import styled from 'styled-components';
import { rem } from 'polished';
import { TextLight, MidnightBlue } from '../../colors';

type Props = {
  imgUrl: string;
};

type HexProps = {
  size: number;
  borderSize: number;
};

const Hexagon = styled.div`
  clip-path: polygon(5% 25%, 50% 0, 95% 25%, 95% 75%, 50% 100%, 5% 75%);
`;

const HexOuterBorder = styled(Hexagon)<HexProps>`
  position: relative;
  --s: ${(props) => rem(props.size + 4 * props.borderSize)};
  width: var(--s);
  height: calc(var(--s) * 1);
  background-color: ${TextLight};
`;
const HexInnerBorder = styled(Hexagon)<HexProps>`
  --s: ${(props) => rem(props.size + 2 * props.borderSize)};
  position: absolute;
  top: ${(props) => rem(props.borderSize)};
  left: ${(props) => rem(props.borderSize)};

  width: var(--s);
  height: calc(var(--s) * 1);
  background: ${MidnightBlue};
`;

const HexBadge = styled(Hexagon)<{ url: string } & HexProps>`
  --s: ${(props) => rem(props.size)};
  position: absolute;
  top: ${(props) => rem(props.borderSize)};
  left: ${(props) => rem(props.borderSize)};

  width: var(--s);
  height: calc(var(--s) * 1);
  display: inline-block;
  font-size: initial; /* we reset the font-size if we want to add some content */
  background: url('${(props) => props.url}') no-repeat center center;
  background-size: cover;
`;

export const GitPOAPBadge = ({ imgUrl }: Props) => {
  const size = 200;
  const borderSize = 4;
  return (
    <HexOuterBorder size={size} borderSize={borderSize}>
      <HexInnerBorder size={size} borderSize={borderSize}>
        <HexBadge url={imgUrl} size={size} borderSize={borderSize} />
      </HexInnerBorder>
    </HexOuterBorder>
  );
};

/* Another option is to use SVG ~ https://stackoverflow.com/questions/67458234/mask-image-with-svg-shape-and-add-a-border*/
