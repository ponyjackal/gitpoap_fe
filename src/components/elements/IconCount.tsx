import styled from 'styled-components';
import { rem } from 'polished';
import { ExtraHover, ExtraPressed, TextGray, TextLight } from '../../colors';

type Props = {
  icon: React.ReactNode;
  href?: string;
  count: number;
};

const Count = styled.span`
  font-family: PT Mono;
  font-style: normal;
  font-weight: bold;
  font-size: ${rem(12)};
  line-height: ${rem(17)};
  letter-spacing: ${rem(0.5)};
  text-transform: uppercase;
  color: ${TextLight};
  margin-left: ${rem(6)};
`;

const Container = styled.a`
  display: inline-flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;

  path {
    transition: 200ms fill ease;
    fill: ${TextGray};
  }

  &:hover {
    path {
      fill: ${ExtraHover};
    }
  }
  &:active {
    path {
      fill: ${ExtraPressed};
    }
  }
`;

export const IconCount = ({ icon, href, count }: Props) => {
  return (
    <Container href={href}>
      {icon}
      {count && <Count>{count}</Count>}
    </Container>
  );
};
