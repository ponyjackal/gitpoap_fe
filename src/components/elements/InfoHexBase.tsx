import React from 'react';
import styled from 'styled-components';
import { rem } from 'polished';
import { BackgroundPanel } from '../../colors';

type Props = {
  children: React.ReactNode;
};

const StyledInfoHex = styled.div`
  display: inline-flex;
  flex-direction: column;
  min-width: ${rem(250)};
`;

const UpperTriangle = styled.div`
  clip-path: polygon(50% 0%, 0% 100%, 100% 100%);
  width: 100%;
  height: ${rem(70)};
  background-color: ${BackgroundPanel};
`;

const LowerTriangle = styled(UpperTriangle)`
  clip-path: polygon(50% 100%, 0 0, 100% 0);
`;

const Body = styled.div`
  background-color: ${BackgroundPanel};
  min-height: ${rem(100)};
`;

export const InfoHexBase = (props: Props) => {
  return (
    <StyledInfoHex>
      <UpperTriangle />
      <Body>{props.children}</Body>
      <LowerTriangle />
    </StyledInfoHex>
  );
};
