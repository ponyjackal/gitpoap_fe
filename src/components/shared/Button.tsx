import React from 'react';
import styled from 'styled-components';
import { Button as MButton } from '@mantine/core';
import { rem } from 'polished';
import { PrimaryBlue, TextAccent } from '../../colors';

type Props = {
  children: React.ReactNode;
  className?: string;
  onClick?: React.MouseEventHandler;
  disabled?: boolean;
  leftIcon?: React.ComponentProps<typeof MButton>['leftIcon'];
};

const StyledButton = styled(MButton)`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  background-color: ${PrimaryBlue};
  font-family: 'PT Mono', monospace;
  font-weight: 700;
  color: white;
  text-transform: uppercase;
  font-size: ${rem(12)};
  line-height: ${rem(18)};
  letter-spacing: ${rem(2)};
  transition: 200ms background ease;
  border-radius: ${rem(6)};
  padding: ${rem(8)} ${rem(14)};
  height: auto;

  &:hover {
    box-shadow: none;
    background-color: ${TextAccent};
  }

  &:active {
    background-color: #20457e;
  }
`;

export const Button = ({ children, className, onClick, disabled, leftIcon }: Props) => {
  return (
    <StyledButton
      compact
      className={className}
      size="md"
      disabled={disabled}
      onClick={onClick}
      leftIcon={leftIcon}
    >
      {children}
    </StyledButton>
  );
};
