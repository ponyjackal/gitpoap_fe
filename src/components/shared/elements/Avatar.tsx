import styled from 'styled-components';
import { rem } from 'polished';

export const Avatar = styled.img`
  height: ${rem(80)};
  width: ${rem(80)};
  border-radius: 50%;
`;
