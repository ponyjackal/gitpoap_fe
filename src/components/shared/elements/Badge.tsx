import styled from 'styled-components';
import { rem } from 'polished';
import { BackgroundPanel3, TextLight } from '../../../colors';

export const Badge = styled.div`
  display: inline-flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: ${rem(0.5)} ${rem(4)};
  background: ${BackgroundPanel3};
  border-radius: ${rem(4)};

  font-family: PT Mono;
  font-style: normal;
  font-weight: bold;
  font-size: ${rem(11)};
  line-height: ${rem(18)};
  letter-spacing: ${rem(1.2)};
  text-transform: uppercase;
  color: ${TextLight};
`;
