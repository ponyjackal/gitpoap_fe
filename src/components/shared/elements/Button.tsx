import { Button as ButtonUI, ButtonProps } from '@mantine/core';
import React from 'react';

type Props = ButtonProps & {
  className?: string;
  onClick?: React.MouseEventHandler;
  onMouseEnter?: React.MouseEventHandler;
  onMouseLeave?: React.MouseEventHandler;
  id?: string;
};

export const Button = (props: Props) => {
  return <ButtonUI {...props}>{props.children}</ButtonUI>;
};
