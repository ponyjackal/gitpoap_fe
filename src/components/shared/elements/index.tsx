export * from './Avatar';
export * from './Badge';
export * from './Button';
export * from './Checkbox';
export * from './ClaimCircle';
export * from './CollapsibleAddress';
export * from './CopyableText';
export * from './DateInput';
export * from './Divider';
export * from './GitPOAPBadge';
export * from './Header';
export * from './HexagonPath';
export * from './IconCount';
export * from './InfoHexBase';
export * from './Input';
export * from './InputWrapper';
export * from './Label';
export * from './Loader';
export * from './MultiSelect';
export * from './NavLink';
export * from './NumberInput';
export * from './OrgName';
export * from './Pagination';
export * from './POAPBadge';
export * from './Radio';
export * from './RelativeDate';
export * from './Select';
export * from './Share';
export * from './Skeletons';
export * from './Text';
export * from './TextArea';
export * from './Title';
export * from './TwitterShareButton';
