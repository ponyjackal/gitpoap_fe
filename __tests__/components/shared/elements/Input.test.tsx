import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import 'jest-styled-components';
import { Input } from '../../../../src/components/shared/elements';

describe('Input', () => {
  it('renders an Input', async () => {
    const spy = jest.fn();

    const { container } = render(<Input onChange={spy} value="" placeholder="placeholder text" />);
    const inputWrapper = container.firstChild;
    expect(inputWrapper).toMatchSnapshot();

    const input = screen.getByRole('textbox');
    expect(screen.getByRole('textbox')).toHaveValue('');
    expect(input).toBeInTheDocument();
    expect(input).toHaveAttribute('placeholder', 'placeholder text');

    await userEvent.type(screen.getByRole('textbox'), 'test-value');
    expect(spy).toHaveBeenCalled();
  });
});
