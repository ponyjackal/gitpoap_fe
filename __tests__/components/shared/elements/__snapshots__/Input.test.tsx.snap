// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders an Input 1`] = `
.c0 {
  display: inline-block;
}

.c0 .mantine-TextInput-label {
  font-family: 'PT Mono',monospace;
  color: #E2E2EE;
  font-weight: 700;
  font-size: 0.6875rem;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.075rem;
  -moz-letter-spacing: 0.075rem;
  -ms-letter-spacing: 0.075rem;
  letter-spacing: 0.075rem;
  line-height: 1.125rem;
  margin-bottom: 0.6875rem;
}

.c0 .mantine-TextInput-input {
  border-radius: 0.375rem;
  font-family: 'PT Mono',monospace;
  font-weight: 400;
  font-size: 0.875rem;
  color: white;
  background-color: #1E1F2E;
  -webkit-transition: 150ms background-color ease;
  transition: 150ms background-color ease;
  -webkit-letter-spacing: 0.0125rem;
  -moz-letter-spacing: 0.0125rem;
  -ms-letter-spacing: 0.0125rem;
  letter-spacing: 0.0125rem;
  border: 0.0625rem solid transparent;
}

.c0 .mantine-TextInput-input:hover:not(:disabled):not(:focus) {
  background-color: #292B3D;
}

.c0 .mantine-TextInput-input:focus {
  background-color: #1E1F2E;
  border: 0.0625rem solid #393A51 !important;
}

.c0 .mantine-TextInput-input::-webkit-input-placeholder {
  color: #393A51;
}

.c0 .mantine-TextInput-input::-moz-placeholder {
  color: #393A51;
}

.c0 .mantine-TextInput-input:-ms-input-placeholder {
  color: #393A51;
}

.c0 .mantine-TextInput-input::placeholder {
  color: #393A51;
}

.c0 .mantine-TextInput-input.mantine-TextInput-invalid {
  color: #F77070;
}

.c0 .mantine-TextInput-error {
  font-family: 'PT Mono',monospace;
  font-size: 0.625rem;
  -webkit-letter-spacing: 1.2px;
  -moz-letter-spacing: 1.2px;
  -ms-letter-spacing: 1.2px;
  letter-spacing: 1.2px;
  padding-left: 1rem;
  color: #F77070 !important;
}

.emotion-2 {
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  line-height: 1.55;
}

.emotion-5 {
  position: relative;
}

.emotion-8 {
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  height: 36px;
  -webkit-tap-highlight-color: transparent;
  line-height: 34px;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  resize: none;
  box-sizing: border-box;
  font-size: 14px;
  width: 100%;
  color: #000;
  display: block;
  text-align: left;
  min-height: 36px;
  padding-left: 12px;
  padding-right: 12px;
  border-radius: 4px;
  border: 1px solid #ced4da;
  background-color: #fff;
  -webkit-transition: border-color 100ms ease;
  transition: border-color 100ms ease;
}

.emotion-8:disabled {
  background-color: #f1f3f5;
  color: #909296;
  opacity: 0.6;
  cursor: not-allowed;
}

.emotion-8:disabled::-webkit-input-placeholder {
  color: #909296;
}

.emotion-8:disabled::-moz-placeholder {
  color: #909296;
}

.emotion-8:disabled:-ms-input-placeholder {
  color: #909296;
}

.emotion-8:disabled::placeholder {
  color: #909296;
}

.emotion-8::-webkit-input-placeholder {
  opacity: 1;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #adb5bd;
}

.emotion-8::-moz-placeholder {
  opacity: 1;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #adb5bd;
}

.emotion-8:-ms-input-placeholder {
  opacity: 1;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #adb5bd;
}

.emotion-8::placeholder {
  opacity: 1;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #adb5bd;
}

.emotion-8::-webkit-inner-spin-button,
.emotion-8::-webkit-outer-spin-button,
.emotion-8::-webkit-search-decoration,
.emotion-8::-webkit-search-cancel-button,
.emotion-8::-webkit-search-results-button,
.emotion-8::-webkit-search-results-decoration {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
}

.emotion-8[type=number] {
  -moz-appearance: textfield;
}

.emotion-8:focus,
.emotion-8:focus-within {
  outline: none;
  border-color: #228be6;
}

<div
  class="emotion-0 emotion-1 c0 emotion-2"
>
  <div
    class="emotion-3 emotion-4 emotion-5"
  >
    <input
      aria-invalid="false"
      class="emotion-6 emotion-7 emotion-8"
      id="mantine-r0"
      placeholder="placeholder text"
      spellcheck="false"
      type="text"
      value=""
    />
  </div>
</div>
`;
